python/hail/docs/expr/hailType.rst

```diff
@@ -0,0 +1,11 @@
+This is the representation of and documentation for a Hail type. Values of this
+type have two representations: the expression language representation and the
+Python representation. This is confusing! There are two sets of documentation
+for many objects like Variant and Genotype. In methods like
+:py:meth:`~hail.VariantDataset.annotate_variants_expr` and 
```

Comment:
I think this is too much.  You introduced the two kinds of values, then you went on to discuss various kinds of functions.  I would just say there are two kinds of values, in expressions and in Python.  Then link to the two kinds.  Will also make the transition to the bulleted lists smoother.